.github/workflows/on-commit.yml

```
@@ -31,51 +39,40 @@ jobs:
             set -xe
             mount --bind /cvasl/test_data/ /ci/test_data/
             cd /ci
-            python -m pip install pycodestyle "pytest<8.1" nbmake
-            python ./setup.py bdist_wheel
-            python -m pip install ./dist/*.whl
-            python ./setup.py test'
-
+            python -m pip install tox
+            python -m tox -e  ${{ matrix.tox-python }} \
+            -- tests/test_core.py'
   SanityOnMulti:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: ['3.10', '3.11']
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: recursive
-      - uses: s-weigand/setup-conda@v1
-        with:
-          conda-channels: conda-forge
-          python-version: '3.11'
-      - run: python3 -m venv .venv
-      - run: .venv/bin/python -m pip install pycodestyle "pytest<8.1" nbmake
-      - run: .venv/bin/python ./setup.py bdist_wheel
-      - run: .venv/bin/python -m pip install ./dist/*.whl
-      - run: .venv/bin/python ./setup.py test -a "test_multi.py"
-      # needs to be updated to include Windows
+      runs-on: ${{ matrix.os }}
+      strategy:
+        matrix:
+          os: [ubuntu-latest, macos-latest, windows-latest]
+          python-version: ['3.10', '3.11']
+          include:
+            - python-version: '3.10'
+              tox-python: "py310"
+            - python-version: '3.11'
+              tox-python: 'py311'
+            - os: windows-latest
+              python: .venv/Scripts/python.exe
+            - os: ubuntu-latest
+              python: .venv/bin/python
+            - os: macos-latest
+              python: .venv/bin/python
+      steps:
+        - uses: actions/checkout@v3
+          with:
+            submodules: recursive
+        - uses: s-weigand/setup-conda@v1
+          with:
+            conda-channels: conda-forge
+            python-version: ${{ matrix.python-version }}
+        - run: python3 -m venv .venv
+        - run: ${{ matrix.python }} -m pip install tox
+        - run: >
+            ${{ matrix.python }} -m tox -e ${{ matrix.tox-python }}
+            -- tests/test_multi.py
```

See explanation in comment above.

```suggestion
        - run: python -m venv .venv
        - run: source .venv/bin/activate
          if: runner.os != 'Windows'
        - run: .venv\Scripts\activate.bat
          if: runner.os == 'Windows'
        - run: python -m pip install tox
        - run: python -m tox -e ${{ matrix.tox-python }} -- tests/test_multi.py
```

Note that I'm not sure whether GitHub Actions runs on cmd.exe or PowerShell, if the latter then you need Activate.ps1 instead of activate.bat.

Note also that I also changed `python3` to `python`. There were some issues in the Python template that were solved by this, I don't remember exactly which, I can look it up for you later if you want (or just browse the closed issues on the template repo). It had something to do with running on Windows I think.

---

This has the same issue as above.